docs/decisions/0002-redux-tests.rst

```diff
@@ -0,0 +1,102 @@
+2. Redux Store Testing
+--------------------------------
+
+Status
+------
+
+Accepted
+
+Context
+-------
+
+The frontend-app-payment repository is a critical piece of infrastructure (the checkout page), and thus has higher testing requirements than some of our other micro-frontends.  This ADR describes a method of writing Jest tests for the reducers, actions, and selectors that exercise the contracts between them.
+
+Decision
+--------
+
+Each sub-module should have a redux.test.js file which exercises its reducers, actions, and selectors.
+
+Method
+======
+
+The test should accomplish this by creating a redux store for the reducers.  If there are selectors to be tested, the reducer should be mounted to the store in the location expected by the selectors, as shown below.  (This is because the selectors are aware of the complete 'path' to the data in the global store and depend on being able to traverse it.)
+
+If there are no selectors to be tested, the below can be simplified by creating a store from the reducer-under-test directly.
+
+::
+
+  describe('redux tests', () => {
+    let store;
+    beforeEach(() => {
+
+      const parentReducer = (state, action) => ({
+        payment: {
+          coupon: reducer(state, action),
+        }
+      });
+
+      store = createStore(parentReducer);
+    });
+
+    // tests
+  });
+
+The test can then use Jest's snapshot expectations to assert the states of the reducer after actions have occurred.
+
+::
+
+  it('should match default state', () => {
+    expect(store.getState()).toMatchSnapshot();
+  });
+
+  it('should begin to add a coupon', () => {
+    store.dispatch(addCouponBegin()); // dispatching of an action
+    expect(store.getState()).toMatchSnapshot();
```

Comment:
Do you think snapshots should be preferred to inspecting specific properties in the state?

Reply:
Thought about that - Yeah, given that the whole state is the thing you want to assert.